### PR TITLE
Adding a new ARM Template for restoring a VM from Restore Point

### PR DESCRIPTION
**Pull request by shahja1**

### CODE REVIEW

File: RestoreVMFromRestorePoint.json [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        },
+        "datadiskRestorePointResourceId": {
+            "type": "string",
+            "defaultValue": "/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Compute/restorePointCollections/<RestorePointCollectionName>/restorePoints/<RestorePointName>/diskRestorePoints/<DataDiskRestorePointName>"
+        },
+        "restoredOSDiskName": {
+            "type": "string",
+            "defaultValue": "<restored_osdisk_name>"
+        },
+        "restoredDataDiskName": {
```
Comment (dikethir) [inline RestoreVMFromRestorePoint.json:61]: Customer can change the type (standard, premium, LRS, ZRS) and size (1TB, 4TB etc. ) of the disk right when restoring from diskRestorePoint right? Can we add this to the template?

Comment (shahja1) [reply RestoreVMFromRestorePoint.json:61]: Yes, they can change these params. Do you want me to parameterize these?

Comment (dikethir) [reply RestoreVMFromRestorePoint.json:61]: Yes please.

File: RestoreVMFromRestorePoint.json [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        },
+        "virtualMachineComputerName": {
+            "type": "string",
+            "defaultValue": "<computeName>"
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "<vmSize>"
+        },
+        "adminUsername": {
```
Comment (prchin) [inline RestoreVMFromRestorePoint.json:41]: Shall we also add osProfile to use these?

File: RestoreVMFromRestorePoint.json [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            "defaultValue": "Premium_LRS"
+        },
+        "restoredDataDiskSizeGB": {
+            "type": "string",
+            "defaultValue": "1024"
+        }
+    },
+    "variables": {
+        "nsgId": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups', parameters('networkSecurityGroupName'))]",
+        "vnetName": "[parameters('virtualNetworkName')]",
```
Comment (prchin) [inline RestoreVMFromRestorePoint.json:84]: Not required?

Comment (shahja1) [reply RestoreVMFromRestorePoint.json:84]: Yes, not required. Doesn't break anything to have it either. ;)

File: RestoreVMFromRestorePoint.json (hunk 2 above)
Comment (shahja1) [reply RestoreVMFromRestorePoint.json:41]: osProfile is not allowed for "Attach OSDisk" scenario, since the OSDisk is already specialized.

**Review by dikethir — APPROVED** [review]
Looks good to be me

**Review by prchin — APPROVED** [review]

**Pull request merged**